vm/actor/src/builtin/market/mod.rs

```diff
@@ -4,7 +4,687 @@
+    if deal.proposal.storage_price_per_epoch < min_price
+        || deal.proposal.storage_price_per_epoch > max_price
+    {
+        return Err(ActorError::new(
+            ExitCode::ErrIllegalArgument,
+            "Storage price out of bounds.".to_owned(),
+        ));
+    };
+
+    let (min_provider_collateral, max_provider_collateral) =
+        deal_provider_collateral_bounds(deal.proposal.piece_size, deal.proposal.duration());
+    if deal.proposal.provider_collateral < min_provider_collateral
+        || deal.proposal.provider_collateral > max_provider_collateral
+    {
+        return Err(ActorError::new(
+            ExitCode::ErrIllegalArgument,
+            "Provider collateral out of bounds.".to_owned(),
+        ));
+    };
+
+    let (min_client_collateral, max_client_collateral) =
+        deal_client_collateral_bounds(deal.proposal.piece_size, deal.proposal.duration());
+    if deal.proposal.provider_collateral < min_client_collateral
+        || deal.proposal.provider_collateral > max_client_collateral
+    {
+        return Err(ActorError::new(
+            ExitCode::ErrIllegalArgument,
+            "Client collateral out of bounds.".to_owned(),
+        ));
+    };
+
+    Ok(())
+}
+
+fn deal_proposal_is_internally_valid<BS, RT>(
+    rt: &RT,
+    proposal: &ClientDealProposal,
+) -> Result<(), ActorError>
+where
+    BS: BlockStore,
+    RT: Runtime<BS>,
+{
+    if proposal.proposal.end_epoch <= proposal.proposal.start_epoch {
+        return Err(ActorError::new(
+            ExitCode::ErrIllegalArgument,
+            "proposal end epoch before start epoch".to_owned(),
+        ));
+    }
+    // Generate unsigned bytes
+    let sv_bz = to_vec(&proposal.proposal).map_err(|_| {
+        rt.abort(
+            ExitCode::ErrIllegalArgument,
+            "failed to serialize DealProposal",
+        )
+    })?;
+
+    rt.syscalls()
+        .verify_signature(
+            &proposal.client_signature,
+            &proposal.proposal.client,
+            &sv_bz,
+        )
+        .map_err(|e| {
+            rt.abort(
+                ExitCode::ErrIllegalArgument,
+                format!("signature proposal invalid: {}", e),
+            )
+        })?;
+
+    Ok(())
+}
+
+// Resolves a provider or client address to the canonical form against which a balance should be held, and
+// the designated recipient address of withdrawals (which is the same, for simple account parties).
+fn escrow_address<BS, RT>(rt: &RT, addr: &Address) -> Result<(Address, Address), ActorError>
+where
+    BS: BlockStore,
+    RT: Runtime<BS>,
+{
+    // Resolve the provided address to the canonical form against which the balance is held.
+    let nominal = rt.resolve_address(addr)?;
+
+    let code_id = rt.get_actor_code_cid(&nominal)?;
+
+    if code_id != *MINER_ACTOR_CODE_ID {
+        // Ordinary account-style actor entry; funds recipient is just the entry address itself.
+        rt.validate_immediate_caller_type(CALLER_TYPES_SIGNABLE.iter())?;
```

Comment:
Can you add the TODO to for the `RequestMinerControlAddrs` call equivalent that is left out so this doesn't get missed (maybe you left out because it is connected to Miner Actor?)

Reply:
That is there, I just changed the conditional blocks.